app/src/main/java/com/easyfitness/programs/ProgramRunnerFragment.java

```diff
@@ -122,33 +122,36 @@
-            long profileId=getProfile().getId();
+        if (mRunningProgram == null) {
             mRunningProgram=(Program)mProgramsSpinner.getSelectedItem();
-            ProgramHistory programHistory = new ProgramHistory(-1, runningProgramId, profileId, ProgramStatus.RUNNING, DateConverter.currentDate(), DateConverter.currentTime(), "", "");
-            long workoutHistoryId = mDbWorkoutHistory.add(programHistory);
-            mRunningProgramHistory = mDbWorkoutHistory.get(workoutHistoryId);
-            mProgramsSpinner.setEnabled(false);
-            mStartStopButton.setText(R.string.finish_program);
-
-            // add all template records with status "Pending"
-            Cursor cursor = mDbRecord.getProgramTemplateRecords(((Program)mProgramsSpinner.getSelectedItem()).getId());
-            List<Record> recordList = mDbRecord.fromCursorToList(cursor);
-            for (Record record:recordList)
-            {
-                record.setTemplateRecordId(record.getId());
-                record.setTemplateSessionId(workoutHistoryId);
-                record.setRecordType(RecordType.PROGRAM_RECORD_TYPE);
-                record.setProgramRecordStatus(ProgramRecordStatus.PENDING);
-                record.setProfileId(getProfile().getId());
-                mDbRecord.addRecord(record);
+            if (mRunningProgram != null) {
+                long runningProgramId = mRunningProgram.getId();
+                long profileId = getProfile().getId();
+                ProgramHistory programHistory = new ProgramHistory(-1, runningProgramId, profileId, ProgramStatus.RUNNING, DateConverter.currentDate(), DateConverter.currentTime(), "", "");
+                long workoutHistoryId = mDbWorkoutHistory.add(programHistory);
+                mRunningProgramHistory = mDbWorkoutHistory.get(workoutHistoryId);
+                mProgramsSpinner.setEnabled(false);
+                mStartStopButton.setText(R.string.finish_program);
+
+                // add all template records with status "Pending"
+                Cursor cursor = mDbRecord.getProgramTemplateRecords(((Program) mProgramsSpinner.getSelectedItem()).getId());
+                List<Record> recordList = mDbRecord.fromCursorToList(cursor);
+                for (Record record : recordList) {
+                    record.setTemplateRecordId(record.getId());
+                    record.setTemplateSessionId(workoutHistoryId);
+                    record.setRecordType(RecordType.PROGRAM_RECORD_TYPE);
+                    record.setProgramRecordStatus(ProgramRecordStatus.PENDING);
+                    record.setProfileId(getProfile().getId());
+                    mDbRecord.addRecord(record);
+                }
+                // refresh table
+                refreshData();
             }
-            // refresh table
-            refreshData();
-        } else {
+
+        }else if (mRunningProgram!=null) {
             stopProgram();
+        }else{
```

Comment:
This "Else" will never be called. At least on of the two conditions of your previous **if** "if (mRunningProgram == null)" and **else if** "}else if (mRunningProgram!=null)" will trigger. 
It can be removed.
Thanks.

Reply:
Agree. Thanks